tile/platform/local_machine/mem_cache.h

```diff
@@ -17,11 +18,16 @@ namespace local_machine {
 // Caches device memory allocations.
 class MemCache {
  public:
+  MemCache(const std::shared_ptr<DevInfo>& dev_info, hal::Memory* source);
   std::shared_ptr<hal::Buffer> TryAlloc(std::size_t size);
   void Free(std::size_t size, std::shared_ptr<hal::Buffer>);
+  void Cleanup(size_t new_size);
 
  private:
+  size_t total_;
+  size_t mem_limit_;
```

Comment:
`mem_limit_` should be const.  (I'm not sure it belongs here, but if it does, it should be const.  :-)